lib/scenic/statements.rb

```diff
@@ -110,13 +109,13 @@ def replace_view(name, version: nil, revert_to_version: nil, materialized: false
         raise ArgumentError, "version is required"
       end
 
-      if materialized
-        raise ArgumentError, "Cannot replace materialized views"
-      end
-
       sql_definition = definition(name, version)
 
-      Scenic.database.replace_view(name, sql_definition)
+      if materialized
+        Scenic.database.replace_materialized_view(name, sql_definition)
```

Comment:
I think 
```
Scenic.database.replace_materialized_view(name, sql_definition)
```
should be
```
Scenic.database.update_materialized_view(name, sql_definition)
```
otherwise `IndexReapplication.new` won't be called